lib/ask_web/controllers/respondent_controller.ex

```diff
@@ -1130,29 +1139,21 @@ defmodule AskWeb.RespondentController do
         fn _ -> [] end
       )
 
+    tz_offset_in_seconds = Survey.timezone_offset_in_seconds(survey)
+    tz_offset = Survey.timezone_offset(survey)
+
     csv_rows =
       log_entries
       |> Stream.map(fn e ->
-        channel_name =
-          if e.channel do
-            e.channel.name
-          else
-            ""
-          end
-
-        disposition = disposition_label(e.disposition)
-        action_type = action_type_label(e.action_type)
-        timestamp = csv_datetime(e.timestamp, survey)
-
         [
           Integer.to_string(e.id),
           e.respondent_hashed_number,
           interactions_mode_label(e.mode),
```

Comment:
Just a suggestion here... Since you're saying that SQL goes fast and the problem is the processing... could `interactions_mode_label`, `disposition_label`, `action_type_label` be converted in a `CASE ... WHEN` in the query and report some speeding up? The query will be a little bit uglier, but maybe...

Reply:
I tried to inline the calls, but I couldn't get any noticeable speedup. I also tried to be more idiomatic Erlang, with multiple definitions and using argument matchers, but it also didn't bring any improvement.

```elixir
  defp interactions_mode_label("mobileweb"), do: "Mobile Web"
  defp interactions_mode_label(mode), do: String.upcase(mode)

  defp action_type_label(nil), do: nil
  defp action_type_label("contact"), do: "Contact attempt"
  defp action_type_label(action), do: String.capitalize(action)

  defp disposition_label(nil), do: nil
  defp disposition_label(disposition), do: String.capitalize(disposition)
```